sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -1114,6 +1114,9 @@ class Analyzer(override val catalogManager: CatalogManager) extends RuleExecutor
           case _ => write
         }
 
+      case PlanWithUnresolvedIdentifier(expr, builder) if expr.resolved =>
```

Comment:
shall we add a new rule to resolve these identifier related plans/expressions?

Reply:
If "we" means @gengliangwang or @cloud-fan ... 👍